lib/security/security-spec.js

```diff
@@ -0,0 +1,70 @@
+var assert = require('assert');
+var sampleUserProfileData = require('../../test/fixtures/sampleUserProfile.json');
+var aes = require('./aes.js');
+var sha256 = require('./sha256.js');
+var sampleSecurityeData = require('../../test/fixtures/sampleSecurityData.json');
+
+describe("Test Data Encyption, Hashing and Decryption", function() {
+  it('should error when you try to hash a null value', function(done) {
+    return sha256.hash(null).then(function(response) {
+      assert(!response, "Should not contain a hash value,");
+    }).then(done).catch(function(err) {
+      assert(err, "should contain an error.");
+      done();
+    });
+  });
+
+  it('should respond with a correct hash when you try to hash a string value', function(done) {
+    sha256.hash(sampleSecurityeData.plaintext).then(function(response) {
+      assert.equal(response, sampleSecurityeData.plaintextSha256);
+    }).then(done);
+  });
+
+  it('should error when you try to encrypt using a null data value', function(done) {
+    aes.encrypt(null, sampleSecurityeData.aesKey).then(function() {
+      done(new Error('It should not try to encrypt using a null data value'));
+    }, function(err) {
```

Comment:
No, he absolutely needs to throw an error in the success case, since it's NOT supposed to succeed, and if it does, the test needs to fail.